src/plugins/data/public/ui/search_bar/search_bar.tsx

```diff
@@ -497,6 +496,7 @@ class SearchBarUI extends Component<SearchBarProps, State> {
           screenTitle={this.props.screenTitle}
           onSubmit={this.onQueryBarSubmit}
           indexPatterns={this.props.indexPatterns}
+          dataSource={this.props.dataSource}
```

Comment:
quick question: do we know if this will have to be reset if toggling this feature off?

like if the select a datasource and toggle it off and that data source isn't available that should be fine right? or do we think in the future we will need to reset it like how i do here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/settings/settings.ts#L52.

this 100% makes sense to me as you have it right now so I don't think we should change this because it's being passed in like index Patterns. but if we do think there might be implications on the toggle on and then off again, might be worth renaming the UI Settings service to something to else to be more descriptive and reset all that stuff in that service.

Reply:
i'm not sure what the behavior would be, i wasn't able to test with data source too much. will need to revisit when we have the test environment